webapp/src/components/rhs/rhs_new_tab.tsx

```diff
@@ -83,14 +87,48 @@ const setEditorText = (text: string) => {
     }
 };
 
-const RHSNewTab = ({botChannelId, selectPost, setCurrentTab}: Props) => {
+const RHSNewTab = ({selectPost, setCurrentTab, activeBot}: Props) => {
     const intl = useIntl();
-
-    // Compatibility with pre v10 create post export
     const dispatch = useDispatch();
-
-    // Compatibility with pre v10 create post export
     const [draft, updateDraft] = useState<any>(null);
+    const [creatingChannel, setCreatingChannel] = useState(false);
+    const currentUserId = useSelector((state: any) => state.entities.users.currentUserId);
+    const botChannelId = activeBot?.dmChannelID || '';
+
+    const currentBots = useSelector((state: any) =>
+        state[`plugins-${manifest.id}`]?.bots || [],
+    );
+
+    // If botChannelId is empty, we need to create a direct channel
+    useEffect(() => {
+        const createDirectChannel = async () => {
+            if (!botChannelId && !creatingChannel && activeBot) {
+                setCreatingChannel(true);
+                const botId = activeBot.id;
+
+                // This will as a side effect create the direct channel for us
+                const newChannelID = await getBotDirectChannel(currentUserId, botId);
```

Comment:
If this fails, I think we'll be left with an infinite spinner. Could you add a catch, and handle that case? Even just a generic "an error has occurred please try again"?